plugins/module_utils/redfish_utils.py

```diff
@@ -3520,3 +3520,119 @@ def delete_volumes(self, storage_subsystem_id, volume_ids):
+        response = self.check_attribute_registry_uri(reg_data, reg_uri)
+        if not response["ret"]:
+            return response
+
+        resp_data, resp_uri = response["resp_data"], response["resp_uri"]
+        # Get the location URI
+        response = self.check_location_uri(resp_data, resp_uri)
+        if not response["ret"]:
+            return response
+
+        rsp_data, rsp_uri = response["rsp_data"], response["rsp_uri"]
+
+        if "RegistryEntries" not in rsp_data:
+            return {
+                "msg": "'RegistryEntries' not present in %s response, %s" % (rsp_uri, str(rsp_data)),
+                "ret": False
+            }
+
+        return {
+            "bios_registry": rsp_data,
+            "bios_registry_uri": rsp_uri,
+            "ret": True
+        }
+
+    def check_attribute_registry_uri(self, reg_data, reg_uri):
+        # Get BiosAttributeRegistry<BIOS version> response
+        resp_data = ""
+        for mem in reg_data['Members']:
+            if "attribute" in mem['@odata.id'].lower():
+                resp = self.get_request(self.root_uri + mem["@odata.id"])
+                if not resp["ret"]:
+                    return resp
+                resp_data = resp["data"]
+                if 'Location' not in resp_data:
+                    msg = "Key 'Location' not found in BIOS attribute registry, URI: %s response: %s"
+                    return {
+                        "ret": False,
+                        "msg": msg % (mem['@odata.id'], str(resp_data))
+                    }
+                break
+        if not resp_data:
+            msg = "'BiosAttributeRegistry<BIOS_version>' URI not found in %s response, %s"
+            return {
+                "ret": False,
+                "msg": msg % (reg_uri, str(reg_data))
+            }
+
+        return {
+            "ret": True,
+            "resp_data": resp_data,
+            "resp_uri": mem['@odata.id']
+        }
+
+    def check_location_uri(self, resp_data, resp_uri):
+        # Get the location URI response
+        # return {"msg": self.creds, "ret": False}
+        rsp_uri = ""
+        for loc in resp_data['Location']:
+            if loc['Language'] == "en":
+                if type(loc['Uri']) is dict and "extref" in loc['Uri'].keys():
+                    rsp_uri = loc['Uri']['extref']
+                    headers = {"Accept": "application/json"}
+                else:
+                    rsp_uri = loc['Uri']
+                    headers = {"Accept": "application/json", "Accept-Encoding": "gzip"}
+        if not rsp_uri:
+            msg = "Language 'en' not found in BIOS Atrribute Registries location, URI: %s, response: %s"
+            return {
+                "ret": False,
+                "msg": msg % (resp_uri, str(resp_data))
+            }
+
+        headers['X-Auth-Token'] = self.creds["token"]
+        res = requests.get(self.root_uri + rsp_uri, headers=headers, verify=False)
+        if res.status_code != 200:
```

Comment:
This should just be:

```
        if res['ret'] is False:
            return res
```

Reply:
This is because of previous issue, when headers are provided we can't use "get_request" method for API call, so will be calling the request module directly where status code will be returned instead of True/False

Reply:
You need to replace that call with `self.get_request(self.root_uri + rsp_uri)`. The requests module cannot be used here.

Reply:
Agree, can we change the "get_request" function to take user defined headers also?
Example: self.get_request(self.root_uri + rsp_uri, "<headers body>")
So that we can use the same function with and without headers

Reply:
You could change `def get_request(self, uri):` to `def get_request(self, uri, override_headers=None):` and then add `if override_headers: req_headers.update(override_headers)`, or something like that. If you need to remove headers, add a `remove_headers` list (default value `None`) that allows to remove headers from `req_headers`, or something like that.

Reply:
Thank you felix, sure we will do that.

Reply:
There shouldn't be a need to override headers at all though; `ef get_request(self, uri)` already takes care of adding user credentials whether it's an X-Auth-Token or HTTP Basic auth.

Reply:
@mraineri and @felixfontein , I have changed code to remove "import request" and also added workaround for HPE servers.